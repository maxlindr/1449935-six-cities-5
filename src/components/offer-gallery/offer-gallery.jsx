import React from 'react';
import PropTypes from 'prop-types';

const MAX_PHOTOS_COUNT = 6;

function OfferGallery(props) {
  return (
    <div className="property__gallery-container container">
      <div className="property__gallery">
        {props.photos.slice(0, MAX_PHOTOS_COUNT).map((url, i) => (
          <div key={url + i} className="property__image-wrapper">
            <img className="property__image" src={url} alt="Photo studio" />
          </div>
        ))}
      </div>
    </div>
  );
}

OfferGallery.propTypes = {
  photos: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default React.memo(OfferGallery);
