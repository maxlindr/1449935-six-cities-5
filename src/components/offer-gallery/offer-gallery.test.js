import React from 'react';
import renderer from 'react-test-renderer';
import {OfferGallery} from './offer-gallery';

const PHOTOS = [
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
  `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`
];

test(`OfferGallery должен корректно отображаться`, () => {
  const component = renderer.create(
      <OfferGallery photos={PHOTOS} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
