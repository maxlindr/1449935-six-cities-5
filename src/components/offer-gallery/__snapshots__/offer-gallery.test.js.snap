// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferGallery должен корректно отображаться 1`] = `
<div
  className="property__gallery-container container"
>
  <div
    className="property__gallery"
  >
    <div
      className="property__image-wrapper"
    >
      <img
        alt="Photo studio"
        className="property__image"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg"
      />
    </div>
    <div
      className="property__image-wrapper"
    >
      <img
        alt="Photo studio"
        className="property__image"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
      />
    </div>
    <div
      className="property__image-wrapper"
    >
      <img
        alt="Photo studio"
        className="property__image"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg"
      />
    </div>
    <div
      className="property__image-wrapper"
    >
      <img
        alt="Photo studio"
        className="property__image"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg"
      />
    </div>
    <div
      className="property__image-wrapper"
    >
      <img
        alt="Photo studio"
        className="property__image"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg"
      />
    </div>
    <div
      className="property__image-wrapper"
    >
      <img
        alt="Photo studio"
        className="property__image"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
      />
    </div>
  </div>
</div>
`;
